### PR TITLE
Support Renaming Containers

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
We are seeking to add the ability to rename a container to our APIs. 

#### Related Pull Requests
* https://github.com/LabKey/labkey-api-js/pull/153
* https://github.com/LabKey/labkey-ui-components/pull/1184
* https://github.com/LabKey/labkey-ui-premium/pull/84
* https://github.com/LabKey/platform/pull/4366
* https://github.com/LabKey/inventory/pull/840
* https://github.com/LabKey/biologics/pull/2113
* https://github.com/LabKey/sampleManagement/pull/1806

#### Changes
* Implement and add renameContainer()

### CODE REVIEW

File: src/labkey/security/Container.ts [hunk 1]
```diff
@@ -85,6 +85,35 @@ export function deleteContainer(config: DeleteContainerOptions): XMLHttpRequest
     });
 }
 
+export interface RenameContainerOptions extends RequestCallbackOptions {
+    /** The new container name. If not specified, defaults to existing name. */
+    name?: string;
+    /** The new container title. If not specified, defaults to name. */
+    title?: string;
+    /** If set to true, adds an alias for the container's current name. */
+    addAlias?: boolean;
```
Comment (labkey-nicka) [inline src/labkey/security/Container.ts:90]: nit: These parameters should be declared in alphabetical order.

File: src/labkey/security/Container.ts [hunk 2]
```diff
@@ -85,6 +85,35 @@ export function deleteContainer(config: DeleteContainerOptions): XMLHttpRequest
     });
 }
 
+export interface RenameContainerOptions extends RequestCallbackOptions {
```
Comment (labkey-nicka) [inline src/labkey/security/Container.ts]: So based on my reading of these options you could supply no parameters and receive a successful result?

**Review by labkey-nicka — APPROVED** [review]

File: src/labkey/security/Container.ts (hunk 2 above)
Comment (RosalineP) [reply src/labkey/security/Container.ts]: Either the name or the title is required, and the server-side code handles checking and validation. I thought that doing a more involved type for this situation seemed like overkill—but I have added a comment to make this more clear

**Review by labkey-nicka — APPROVED** [review]

File: src/labkey/security/Container.ts (hunk 2 above)
Comment (labkey-nicka) [inline src/labkey/security/Container.ts]: After manual testing I realized this isn't fully typed for the payload response. Consider:

```ts
export interface RenameContainerOptions extends RequestCallbackOptions<Container>
```

**Pull request merged**